options.go

```diff
@@ -304,13 +304,13 @@ func WithLogHandler(handler slog.Handler) func(*Server) {
 	}
 }
 
-// WithSerializer sets a custom serializer that overrides the default one.
+// WithSerializer sets a custom serializer of type Sender that overrides the default one.
 // Please send a PR if you think the default serializer should be improved, instead of jumping to this option.
-func WithSerializer(serializer func(w http.ResponseWriter, r *http.Request, ans any) error) func(*Server) {
+func WithSerializer(serializer Sender) func(*Server) {
 	return func(c *Server) { c.Serialize = serializer }
 }
 
-func WithErrorSerializer(serializer func(w http.ResponseWriter, r *http.Request, err error)) func(*Server) {
+func WithErrorSerializer(serializer ErrorSender) func(*Server) {
```

Comment:
Adding a comment to this method, could be like the one you modified for WithSerializer a good thing

Reply:
I cannot tell if you fixed, or if I missed, but it's good now :shrug: